agent/src/beerocks/slave/son_slave_thread.cpp

```diff
@@ -4842,7 +4849,8 @@ bool slave_thread::autoconfig_wsc_add_m1()
     tlvWscM1->primary_device_type_attr().sub_category_id = WSC::WSC_DEV_NETWORK_INFRA_AP;
 
     // encryption support - diffie-helman key-exchange
-    std::copy(dh.pubkey(), dh.pubkey() + dh.pubkey_length(), tlvWscM1->public_key_attr().data);
+    dh = std::make_shared<mapf::encryption::diffie_hellman>();
```

Comment:
make_unique would be more appropriate here IMO. dh is only used within this object, so it is not shared. Of course, the behaviour is the same, but unique_ptr is more explicit IMO.